### PR TITLE
Added NXOS provider for radius_server

### PR DESCRIPTION
**Pull request by jonnytpuppet**

### CODE REVIEW

File: tests/beaker_tests/radius_server/radius_serverlib.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  radius_server { '8.8.8.8':
+    ensure              => 'present',
+    accounting_only     => 'false',
+    acct_port           => '44',
+    auth_port           => '55',
+    authentication_only => 'true',
+    key                 => '44444444',
+    key_format          => '7',
+    retransmit_count    => '3',
+    timeout             => '1',
```
Comment (hunner) [inline tests/beaker_tests/radius_server/radius_serverlib.rb]: Could you add arother "change" test setting -1 values for the properties that accept them and verify that they are removed from the show run output?

File: lib/puppet/provider/radius_server/nxapi.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    elsif val.is_a?(Symbol)
+      val.to_s
+    else
+      val
+    end
+  end
+
+  def validate
+    fail ArgumentError,
+         "This provider does not support the 'hostname' property. The namevar should be set to the IP of the Raidus Server" if @resource[:hostname]
```
Comment (glennmatthews) [inline lib/puppet/provider/radius_server/nxapi.rb]: please wrap to 80 characters width. Also fix 'Raidus' => 'Radius'

File: lib/puppet/provider/radius_server/nxapi.rb [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    fail ArgumentError,
+         "This provider does not support the 'group' property." if @resource[:group]
```
Comment (glennmatthews) [inline lib/puppet/provider/radius_server/nxapi.rb]: With the code as written, if the user specifies multiple unsupported properties, only the first one will be reported as a failure - possibly requiring multiple iterations by the user to correct (fix one property, another one fails, fix that one, yet another fails). I wonder if there could be a cleaner way to report all failures in one go - something like:

```
UNSUPPORTED_PROPS = [:group, :deadtime, :vrf, ...]
invalid = UNSUPPORTED_PROPS & @resource.keys
fail ArgumentError, "This provider does not support the following properties: #{invalid}" unless invalid.empty?
```

File: lib/puppet/provider/radius_server/nxapi.rb [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    else
+      if @property_hash.empty?
+        # create a new Radius Server
+        @radius_server = Cisco::RadiusServer.new(@resource[:name])
+      end
+
+      RADIUS_SERVER_PROPS.each do |puppet_prop, cisco_prop|
+        if @resource[puppet_prop]
+          @radius_server.send("#{cisco_prop}=", munge_flush(@resource[puppet_prop])) \
+            if @radius_server.respond_to?("#{cisco_prop}=")
```
Comment (glennmatthews) [inline lib/puppet/provider/radius_server/nxapi.rb]: Perhaps:
```
if @resource[puppet_prop] && @radius_server.respond_to?("#cisco_prop}=")
  @radius_server.send("#{cisco_prop}=", munge_flush(@resource[puppet_prop]))
end
```

**Comment by jonnytpuppet** [comment]
@glennmatthews comments addressed

**Comment by glennmatthews** [comment]
Looks good to me. Can you confirm Beaker test passes? Will merge once we've merged cisco/cisco-network-node-utils#19

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/radius_server/radius_server_provider_defaults.rb passed in 28.67 seconds
      Test Suite: tests @ 2015-10-21 15:32:47 +0100

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/radius_server/radius_server_provider_defaults.rb passed in 29.95 seconds
      Test Suite: tests @ 2015-10-22 16:12:52 +0100

**Pull request merged**